code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -27,10 +28,72 @@ var/const/MAX_ACTIVE_TIME = 400
 	var/strength = 5
 
 	var/attached = 0
+	var/target_time = 0.5 // seconds
+	var/walk_speed = 2 // tenths of seconds
+	var/nextwalk = 0
+	var/mob/living/carbon/human/target = null
 
 /obj/item/clothing/mask/facehugger/can_contaminate()
 	return 0
 
+/obj/item/clothing/mask/facehugger/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+
+/obj/item/clothing/mask/facehugger/process()
+	healthcheck()
+
+
+// HUGGER MOVEMENT AI
+/obj/item/clothing/mask/facehugger/proc/targetting()
+	spawn while(1)
+		if(stat == DEAD || stat == UNCONSCIOUS)
+			walk(src,0)
+			return
+		if(attached == 0 && stat != DEAD && stat != UNCONSCIOUS)
+			if(target == null || target.stat == DEAD || target.stat == UNCONSCIOUS)
+				findtarget()
+			else
+				followtarget()
+
+		sleep(target_time * 10)
+
+
+/obj/item/clothing/mask/facehugger/proc/findtarget()
+	for(var/mob/living/carbon/human/T in hearers(src,4))
```

Comment:
Can't you just prioritize it so that it checks for humans in range and goes after them if any, then checks for monkeys in range and goes after them if any?